### PR TITLE
Material Template Fixes

### PR DESCRIPTION
**Pull request by MarcSkovMadsen**
I have tested the Material Template and see some issues that I would like to fix

- The header items overflow the header on the right
- The header has no box shadow (as the Vanilla template has)
- There are a lot of scrollbars.
- A little bit more padding/ margin is need.
- Cannot set the main max-width. 
- Add a little bit of gray to the sidebar.

image

I will develop the fixes ASAP. Will change the Material Template to behave as the Vanilla Template.

### CODE REVIEW

**Comment by philippjfr** [comment]
See https://github.com/holoviz/panel/pull/1661

**Comment by philippjfr** [comment]
Sorry misread, thought this was Golden layout as well.

**Comment by philippjfr** [comment]
>There are a lot of scrollbars.

So I don't know what's up with that, I do not see scrollbars, only when I actually try to scroll.

**Comment by philippjfr** [comment]
That's probably specific to my particular browser tbh.

**Comment by philippjfr** [comment]
Actually I can't any browser to show all the scrollbars all the time, is this a windows thing?

**Comment by nghenzi** [comment]
I see the scrollbars in windows too with chrome

**Comment by philippjfr** [comment]
Okay, clearly needs fixing but really I blame windows 😜

**Comment by MarcSkovMadsen** [comment]
The only thing I know is that `overflow: auto` is handle differently by Chrome and Firefox. As far as I remember Chrome shows disabled scrollbars while Firefox only shows the scrollbars when needed. Therefore I always set them to `scroll` instead of `auto` because then they behave the same and I don't need to handle margin/ padding/ width differences between browsers.

**Comment by MarcSkovMadsen** [comment]
image

**Comment by MarcSkovMadsen** [comment]
FYI. I will start working on this.

**Comment by MarcSkovMadsen** [comment]
I've now implemented the fixes needed and aligned with the VanillaTemplate.

I have some ideas on providing a MaterialTheme for Bokeh. I will try to add that as well.

**Comment by MarcSkovMadsen** [comment]
I've now added a Bokeh MaterialTheme and MaterialDarkTheme but then 

- the default.css is no longer imported (and it gets the purple header as shown below)
- and I don't think the bokeh theme is actually changed. The font in the plots does not look bigger and Roboto like

image

File: panel/template/theme.py [hunk 1]
```diff
@@ -59,3 +59,113 @@ class DarkTheme(Theme):
 
     bokeh_theme = param.ClassSelector(class_=(_BkTheme, str),
                                       default=_BkTheme(json=BOKEH_DARK))
+
+MATERIAL_FONT = "Roboto, sans-serif, Verdana"
+MATERIAL_THEME = {
```
Comment (philippjfr) [inline panel/template/theme.py:64]: This should live in panel/template/material/__init__.py

**Comment by philippjfr** [comment]
Again, in the interest of making quick progress here I'll merge and make a new PR. You have access rights for this repo so please push directly to holoviz/panel rather than your own fork so we can collaborate more easily.

File: panel/template/theme.py (hunk 1 above)
Comment (MarcSkovMadsen) [reply panel/template/theme.py:64]: That's awesome. I just did not know that one. Will move it.

**Comment by MarcSkovMadsen** [comment]
@philippjfr . Did you have any ideas on fixing the below

-----

I've now added a Bokeh MaterialTheme and MaterialDarkTheme but then

- the default.css is no longer imported (and it gets the purple header as shown below)
a- nd I don't think the bokeh theme is actually changed. The font in the plots does not look bigger and Roboto like

**Comment by MarcSkovMadsen** [comment]
Hi @ceball 

I've have heard you have an interest in applications using Material Theme. If you have any ideas, guidance, requirements feel free to comment. Thanks.

**Comment by philippjfr** [comment]
>the default.css is no longer imported (and it gets the purple header as shown below)

That is deliberate, the default.css corresponds to the default theme. I'm still trying to understand why you created the MaterialTheme instead of extending the DefaultTheme but maybe it's just not well documented and that was your intention?

**Comment by MarcSkovMadsen** [comment]
The intention is to make the Bokeh/ HoloViews plot look consistent with the bootstrap theme. This includes the "Roboto" font, some font size changes, some color changes etc.

Maybe I implemented it wrongly. I believe the Bokeh JSON THEME is correct. But maybe I need some guidance on how to move on from there @philippjfr ?

**Comment by MarcSkovMadsen** [comment]
@philippjfr . One temporary solution for getting material themed widgets could be using a small script to replace the bokeh classes with the material classes. Maybe it's as simple as defining the conversions in a dictionary.

What do you think?

The thing I don't know is if I can get the script to execute every time the css class changes. But maybe I can using a MUTATIONOBSERVER.

If it works it could be a really easy and powerful thing.

### POC

image

image

**Comment by MarcSkovMadsen** [comment]
One problem right now is I don't know how to find documentation for MASTER. I can only find documentation for 0.9.7.

Maybe a link to the MASTER documentation in the GITHUB README or the DEVELOPER GUIDE would be nice to have.

**Comment by philippjfr** [comment]
Here's where those builds go: https://pyviz-dev.github.io/panel/

**Comment by xavArtley** [comment]
@philippjfr
when is updated this site?
For what I see the version is  Panel 0.10.0a24.post1+g2b9c98f documentation

**Comment by philippjfr** [comment]
The builds have been broken for a while, trying to fix that. It's usually rebuilt on each dev release or when a commit ending in `[doc-build]` is merged.

**Pull request merged**